### PR TITLE
Add CBMC proof of receiveAndProcessOtaEvent

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of receiveAndProcessOtaEvent function.

Description
-----------
This PR adds files required for the CBMC proof of receiveAndProcessOtaEvent.

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is formatted using Uncrustify.
- [ ] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/receiveAndProcessOtaEvent/receiveAndProcessOtaEvent_harness.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    if( otaInterface != NULL )
+    {
+        otaInterface->os.event.recv = recvEventStub;
+    }
+
+    otaAgent.pOtaInterface = otaInterface;
+    otaAgent.state = state;
+
+    ( void ) receiveAndProcessOtaEvent();
```
Comment (gshvang) [inline test/cbmc/proofs/receiveAndProcessOtaEvent/receiveAndProcessOtaEvent_harness.c:53]: Can we check any parameter or status that this function will be setting if it executes successfully?

File: test/cbmc/source/stubs.c [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    return err;
+}
+
+OtaOsStatus_t startTimerStub( OtaTimerId_t otaTimerId,
+                              const char * const pTimerName,
+                              const uint32_t timeout,
+                              OtaTimerCallback_t callback )
+{
+    OtaOsStatus_t status;
```
Comment (gshvang) [inline test/cbmc/source/stubs.c:48]: Assumption of status should be done here.

File: test/cbmc/source/stubs.c [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    __CPROVER_assert( pTimerName != NULL, "Error: TimerName cannot be NULL" );
+
+    __CPROVER_assert( callback != NULL, "Error: callback should be pointing to a function." );
+
+    return status;
+}
+
+OtaOsStatus_t stopTimerStub( OtaTimerId_t otaTimerId )
+{
```
Comment (gshvang) [inline test/cbmc/source/stubs.c:61]: Assumption of status should be done here.

**Review by gshvang — APPROVED** [review]

File: test/cbmc/proofs/receiveAndProcessOtaEvent/receiveAndProcessOtaEvent_harness.c (hunk 1 above)
Comment (Bujain) [reply test/cbmc/proofs/receiveAndProcessOtaEvent/receiveAndProcessOtaEvent_harness.c:53]: `receiveAndProcessOtaEvent` only calls the respective handler based on the event received by it. This way we cannot check if the `receiveAndProcessOtaEvent` function ran successfully. The closet that we can go is to look at the coverage report indicates how many functions are called and what lines are hit by CBMC.

File: test/cbmc/source/stubs.c (hunk 2 above)
Comment (Bujain) [reply test/cbmc/source/stubs.c:48]: Updated in 251a3f4

File: test/cbmc/source/stubs.c (hunk 3 above)
Comment (Bujain) [reply test/cbmc/source/stubs.c:64]: Updated in 251a3f4

**Comment by gshvang** [comment]
CBMC checks is failing on this PR.

**Review by gshvang — APPROVED** [review]

**Review by cobusve — APPROVED** [review]

**Review by AniruddhaKanhere — APPROVED** [review]

File: test/cbmc/proofs/receiveAndProcessOtaEvent/receiveAndProcessOtaEvent_harness.c [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    OtaState_t state;
+
+    /* Havoc otaAgent to non-deterministically set all the bytes in
+     * the structure. */
+    __CPROVER_havoc_object( &otaAgent );
+
+    /* Assume different states of the Ota Agent.*/
+    __CPROVER_assume( ( state >= OtaAgentStateNoTransition ) && ( state <= OtaAgentStateAll ) );
+
+    otaInterface = ( OtaInterfaces_t * ) malloc( sizeof( OtaInterfaces_t ) );
```
Comment (paulbartell) [inline test/cbmc/proofs/receiveAndProcessOtaEvent/receiveAndProcessOtaEvent_harness.c:47]: otaInterface should be stack allocated or have __CPROVER_havoc_object on it.

Comment (Bujain) [reply test/cbmc/proofs/receiveAndProcessOtaEvent/receiveAndProcessOtaEvent_harness.c:47]: You are right @paulbartell . I have havoc'd otaInterface object

**Review by paulbartell — APPROVED** [review]

**Review by AniruddhaKanhere — APPROVED** [review]

**Pull request merged**